crates/ink/codegen/src/generator/contract.rs

```diff
@@ -53,6 +53,7 @@ impl GenerateCode for Contract<'_> {
         quote! {
             #( #attrs )*
             #vis mod #ident {
+                use ::ink::prelude::{borrow::*, boxed::*, format, string::*, vec::*};
```

Comment:
Why isn't our prelude flat as it is customary for preludes? Then users can just do `prelude::*`. There also seem to be things in the sub modules that we probably don't want to pull in.

Reply:
If we do that. The dev still needs to explicitly specify crates: `vec::Vec` & `string::String` which is not ideal.

In "classic" rust program this data structures are available by default.

Reply:
> If we do that. The dev still needs to explicitly specify crates: vec::Vec & string::String which is not ideal.

If those types are directly inside `prelude` they will be available by just having `use prelude::*`.

> In "classic" rust program this data structures are available by default.

Because the compiler inserts `use std::prelude::v1::*` for you and the types are all in the root (flat):
https://doc.rust-lang.org/stable/std/prelude/v1/index.html


I am just saying we should not have sub modules in our prelude. It should just be a list of re-exports into the root (sorry for shaving the yak here :D).